File: Leetcode/2415 Reverse Odd Levels of Binary Tree.py
from collections import deque


class Solution:
    def reverseOddLevels(self, root):
        queue = deque([root] if root else [])
        index = 0
        while queue:
            for _ in range(len(queue)):
                node = queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            index += 1
            if index % 2 == 0:
                continue
            level = list(queue)
            i, j = 0, len(level) - 1
            while i < j:
                level[i].val, level[j].val = level[j].val, level[i].val
                i += 1
                j -= 1
        return root

File: Leetcode/2415 Reverse Odd Levels of Binary Tree.cs
public class Solution
{
    public TreeNode? ReverseOddLevels(TreeNode root)
    {
        List<TreeNode> q = [root];
        bool reverse = false;
        while (q.Count > 0)
        {
            List<TreeNode> newQ = [];
            foreach (var node in q)
            {
                if (node.left != null) newQ.Add(node.left);
                if (node.right != null) newQ.Add(node.right);
            }
            if (reverse)
            {
                for (int i = 0, j = q.Count - 1; i < j; i++, j--)
                {
                    (q[i].val, q[j].val) = (q[j].val, q[i].val);
                }
            }
            q = newQ;
            reverse = !reverse;
        }
        return root;
    }
}
